x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_search_bar/alerts_search_bar.tsx

```diff
@@ -44,15 +46,22 @@ export function AlertsSearchBar({
   } = useKibana<TriggersAndActionsUiServices>().services;
 
   const [queryLanguage, setQueryLanguage] = useState<QueryLanguageType>('kuery');
-  const { value: dataView, loading, error } = useAlertDataView(featureIds);
-  const {
-    value: aadFields,
-    loading: fieldsLoading,
-    error: fieldsError,
-  } = useRuleAADFields(ruleTypeId);
+  const { dataview, loading } = useAlertDataView(featureIds ?? []);
+  const { aadFields, loading: fieldsLoading } = useRuleAADFields(ruleTypeId);
 
   const indexPatterns =
-    ruleTypeId && aadFields?.length ? [{ title: ruleTypeId, fields: aadFields }] : dataView;
+    ruleTypeId && aadFields?.length ? [{ title: ruleTypeId, fields: aadFields }] : dataview;
+
+  const ruleType = useLoadRuleTypesQuery({
+    filteredRuleTypes: ruleTypeId !== undefined ? [ruleTypeId] : [],
+    enabled: ruleTypeId !== undefined,
```

Comment:
nitpick: seems odd that we're checking for `undefined` so much, so we're saying we should enable this if ruleTypeId an empty string? I think a `!ruleTypeId` check might be better?